### PR TITLE
Change the  Java sdk harness cache timeout for bundle processors to be an hour for streaming pipelines instead of 1 minute.

### LINKED ISSUE #29797: [Bug]: Java BundleProcessorCache evicts all processors for a bundle descriptor id after 1 minute of idleness
### What happened?

A timeout of 1 minute is specified here:
https://github.com/apache/beam/blob/master/sdks/java/harness/src/main/java/org/apache/beam/fn/harness/control/ProcessBundleHandler.java#L923

The key of this cache is the bundle descriptor id (fused stage) and the value is a list of BundleProcessors cached.  This means that if we don't process a stage for a minute all of the bundle processors are destroyed. For long-lived streaming pipelines this is wasteful as stages are processed in parallel and if were previously processed, will likely process again.  Creating a BundleProcessor involves constructing user DoFn and running Setup method so it is non-trivial.

Some improvements:
- increase the timeout for streaming pipelines by default or make it runner configurable
- don't throw away the entire list of cached BundleProcessors on expiration but perhaps just remove the last processor.

### Issue Priority

Priority: 2 (default / most bugs should be filed as P2)

### Issue Components

- [ ] Component: Python SDK
- [X] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [ ] Component: IO connector
- [ ] Component: Beam YAML
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

### PR DESCRIPTION
**Pull request by scwhittle**
Done via an option so that it can be controlled further such as never GC'ing if desired

fixes #29797

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by scwhittle** [comment]
Run Java PreCommit

**Comment by scwhittle** [comment]
R: @kennknowles as you commented earlier on https://github.com/apache/beam/issues/29797

File: sdks/java/core/src/main/java/org/apache/beam/sdk/options/SdkHarnessOptions.java
Comment (kennknowles) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/options/SdkHarnessOptions.java:391]: I suggest `@Hidden` so that this is not shown to users when they do `--help`

**Review by kennknowles — CHANGES_REQUESTED** [review]

**Review by kennknowles — APPROVED** [review]
Approved once the knob is hidden from users. (feel free to merge when green without waiting for me to re-approve)

File: sdks/java/core/src/main/java/org/apache/beam/sdk/options/SdkHarnessOptions.java
Comment (scwhittle) [reply sdks/java/core/src/main/java/org/apache/beam/sdk/options/SdkHarnessOptions.java:392]: Done!

**Comment by scwhittle** [comment]
Failures are unrelated flakes

**Pull request merged**
